superset/utils.py

```diff
@@ -363,7 +363,7 @@ def pessimistic_json_iso_dttm_ser(obj):
 def datetime_to_epoch(dttm):
     if dttm.tzinfo:
         epoch_with_tz = pytz.utc.localize(EPOCH)
-        return (dttm - epoch_with_tz).total_seconds() * 1000
+        return (dttm.astimezone(pytz.UTC) - epoch_with_tz).total_seconds() * 1000
```

Comment:
if it's not too much to ask, this could use a little test to assert the change / bug fix.